### PR TITLE
Add prose for typing rule of try-catch-catch_all.

### PR DESCRIPTION
**Pull request by ioannad**
Fixes related .. todo:: section.

Also changed notation to match the [formal overview document](https://github.com/WebAssembly/exception-handling/blob/main/proposals/exception-handling/Exceptions-formal-overview.md).

### CODE REVIEW

**Review by aheejin — APPROVED** [review]

File: document/core/valid/instructions.rst [hunk 1]
```diff
@@ -1366,26 +1366,48 @@ Control Instructions
...
-.. todo::
-   Add prose.
+* The :ref:`block type <syntax-blocktype>` must be :ref:`valid <valid-blocktype>` as some :ref:`function type <syntax-functype>` :math:`[t_1^\ast] \to [t_2^\ast]`.
+
+* Let :math:`C'` be the same :ref:`context <context>` as :math:`C`, but with the :ref:`result type <syntax-resulttype>` :math:`[t_2^\ast]` prepended to the |CLABELS| vector.
+
+* Under context :math:`C'`,
+  the instruction sequence :math:`\instr_1^\ast` must be :ref:`valid <valid-instr-seq>` with type :math:`[t_1^\ast] \to [t_2^\ast]`.
+
+* Let :math:`C''` be the same :ref:`context <context>` as :math:`C`, but with the :ref:`label <exec-label>` :math:`\LCATCH~[t_2^\ast]` prepended to the |CLABELS| vector.
```
Comment (rossberg) [inline document/core/valid/instructions.rst:1376]:
```suggestion
* Let :math:`C''` be the same :ref:`context <context>` as :math:`C`, but with the :ref:`label type <syntax-labeltype>` :math:`\LCATCH~[t_2^\ast]` prepended to the |CLABELS| vector.
```

File: document/core/valid/instructions.rst [hunk 2]
```diff
@@ -1366,26 +1366,48 @@ Control Instructions
...
+
+* For every :math:`(\CATCH~x~\instr_2^\ast)`:
```
Comment (rossberg) [inline document/core/valid/instructions.rst:1378]:
```suggestion
* For every :math:`x_i` and :math:`\instr_{2i}^\ast` in :math:`(\CATCH~x~\instr_2^\ast)^\ast`:
```

File: document/core/valid/instructions.rst [hunk 3]
```diff
@@ -1366,26 +1366,48 @@ Control Instructions
...
+
+  * The tag :math:`C.\CTAGS[x]` must be defined in the context :math:`C`.
+
+  * Let :math:`[t^\ast] \to []` be its :ref:`tag type <syntax-tagtype>`.
```
Comment (rossberg) [inline document/core/valid/instructions.rst:1382]: This should actually _check_ that the return type is `[]`.
```suggestion
  * Let :math:`[t_{3i}^\ast] \to [t_{4i}^\ast]` be the :ref:`tag type <syntax-tagtype>` :math:`C.\tags[x]`.

  * The :ref:`result type <syntax-resulttype>` :math:`[t_{4i}^\ast]` must be empty.
```

File: document/core/valid/instructions.rst [hunk 4]
```diff
@@ -1366,26 +1366,48 @@ Control Instructions
...
+
+  * Under context :math:`C''`,
+    the instruction sequence :math:`\instr_2^\ast` must be :ref:`valid <valid-instr-seq>` with type :math:`[t^\ast] \to [t_2^\ast]`.
```
Comment (rossberg) [inline document/core/valid/instructions.rst:1385]:
```suggestion
    the instruction sequence :math:`\instr_{2i}^\ast` must be :ref:`valid <valid-instr-seq>` with type :math:`[t_{3i}^\ast] \to [t_2^\ast]`.
```

File: document/core/valid/instructions.rst [hunk 5]
```diff
@@ -1366,26 +1366,48 @@ Control Instructions
...
+
+* If there is a :math:`(\CATCHALL~\instr_3^\ast)`, then:
```
Comment (rossberg) [inline document/core/valid/instructions.rst:1387]:
```suggestion
* If :math:`(\CATCHALL~\instr_3^\ast)^?` is not empty, then:
```

**Review by rossberg — COMMENTED** [review]
Sorry, being a bit late.

**Comment by aheejin** [comment]
@rossberg Thanks for the review! Just in case you'd like to do final check, there also have been several recently landed PRs.

**Comment by rossberg** [comment]
@aheejin, yes, I saw. I'll comment on the ones I find issues with.

**Comment by rossberg** [comment]
Btw, with the introduction of label type, I think all places that say something like "add result type [t*] to C.labels" need to change to say "label type" instead.

**Comment by ioannad** [comment]
> Btw, with the introduction of label type, I think all places that say something like "add result type [t*] to C.labels" need to change to say "label type" instead.

@rossberg created PR #241 with the additional changes requested here, also changing to "label type" I think in all appropriate places, including `ctrl_frame`s.

**Pull request merged**